c10/xpu/XPUFunctions.cpp

```diff
@@ -94,6 +94,9 @@ void initDeviceProperties(DeviceProp* device_prop, int device) {
   device_prop->platform_name =
       raw_device.get_info<device::platform>().get_info<platform::name>();
 
+  device_prop->device_architecture =
+      raw_device.get_info<oneapi::experimental::info::device::architecture>();
+
```

Comment:
What's the motivation to add the property? By the way, it is an experimental feature. Which bundler version does support this feature?

Reply:
It is used in triton backend for Intel's GPU for the kernel optimization. It was ported from here https://github.com/intel/intel-extension-for-pytorch/commit/e22924d77b478e1c4c37d36861e36a6b05b9274c#diff-0e668f3c38309d9cdc6172d37a6213f3ca2d3e1b77331646516a461292592148R260 , so, I believe it is `>=2024.1`. From what I see pytorch is being built here by g++ with linking to intel's sycl headers. That why there is no variable `__INTEL_LLVM_COMPILER` variable used originally...